feat: add autolabel_metadata for both instance-level and data-level annotations (#233)
tier4/tier4_perception_dataset: docs/t4_format_3d_detailed.md

  - "orientation": [Optional[float]] -- Orientation of the arrow shape within the bounding box, in radians. Present only for categories where `has_orientation` is true (e.g., traffic light arrows).
  - "number": [Optional[int]] -- The digit displayed within the bounding box. Present only for categories where `has_number` is true (e.g., numeric traffic lights).
  - "automatic_annotation": [bool] -- Set to True if the annotation is generated entirely by an ML model. Set to False if at least one part of the annotation (either bbox or mask) is manually modified or annotated.

>   - "autolabel_metadata": [Optional[dict]] -- Metadata about the automatic annotation applied to this entire sample_data item (e.g., image or scan). **(Added)**

Author: Metadata describing the automatic annotation process. Required if `automatic_annotation` is `true`. **(Added)**

>   - "autolabel_metadata": [Optional[dict]] -- Metadata about the automatic annotation applied to this entire sample_data item (e.g., image or scan). **(Added)**

Author: ```suggestion
  - "autolabel_metadata": [Optional[dict]] -- Metadata describing the autolabel. Required if `automatic_annotation` is `true`. **(Added)**. **(Added)**
```

  - "category_token": [str] -- Foreign key pointing to the surface category.
  - "mask": [RLE] -- Run length encoding of segmentation mask using the pycocotools package.
  - "automatic_annotation": [bool] -- True if the annotation is fully generated by an ML model; False if any part is manually modified or annotated.

>   - "autolabel_metadata": [Optional[dict]] -- Metadata about the automatic annotation applied to this entire sample_data item (e.g., image or scan). **(Added)**

Author: ```suggestion
  - "autolabel_metadata": [Optional[dict]] -- Metadata describing the autolabel. Required if `automatic_annotation` is `true`. **(Added)**. 
```
